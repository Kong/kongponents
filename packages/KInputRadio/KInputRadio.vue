<template>
  <label class="k-input-label k-inputCheckbox">
    <input
      :checked="isSelected"
      v-bind="$attrs"
      type="radio"
      class="k-input"
      @click="handleClick">
    <slot>{{ label }}</slot>
  </label>
</template>

<script>
export default {
  name: 'KInputRadio',

  model: {
    prop: 'model',
    event: 'change'
  },

  props: {
    model: {
      type: [String, Number, Boolean, Object],
      required: true
    },
    /**
     * Sets whether or not radio is selected
     */
    value: {
      type: [String, Number, Boolean, Object],
      default: 'on'
    },
    /**
     * Sets label text
     */
    label: {
      type: String,
      default: ''
    }
  },

  computed: {
    isSelected () {
      return this.model === this.value
    }
  },

  methods: {
    handleClick () {
      this.$emit('change', this.value)
    }
  }
}
</script>

<style lang="scss" scoped>
  @import '~@kongponents/styles/_variables.scss';
  @import '~@kongponents/styles/forms/_checkbox-radio.scss';
</style>
