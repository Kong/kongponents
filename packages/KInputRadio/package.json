{
  "name": "@kongponents/kinputradio",
  "version": "0.1.0",
  "description": "KInputRadio description here.",
  "main": "KInputRadio.vue",
  "scripts": {
    "test": "jest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Kong/kongponents.git"
  },
  "author": "Kong Inc",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/Kong/kongponents/issues"
  },
  "homepage": "https://github.com/Kong/kongponents#readme"
}
