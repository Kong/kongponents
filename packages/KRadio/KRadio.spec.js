import { mount } from '@vue/test-utils'
import KInputRadio from '@/KRadio/KRadio'

describe('KInputRadio', () => {
  it('matches snapshot', () => {
    const wrapper = mount(KInputRadio)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
