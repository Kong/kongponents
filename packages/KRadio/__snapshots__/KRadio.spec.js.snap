// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KInputRadio matches snapshot 1`] = `<label class="k-radio"><input type="radio" class="k-input"> </label>`;
